### PR TITLE
Initial web project

### PR DESCRIPTION
**Pull request by benagain**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -58,13 +58,13 @@ steps:
...
-    displayName: "Publish - dotnet publish $(__SolutionName__)"
+    displayName: "Publish Apprentice Portal Web"
     inputs:
       command: publish
       publishWebProjects: false #set to false as this setting (which defaults to true) will parse the entire repo for web projects
-      projects: "src/$(__SolutionName__)/$(__SolutionName__).csproj"
+      projects: src/**/*.csproj
       arguments: "-o $(build.artifactstagingdirectory)/publish -c $(BuildConfiguration) --no-build"
-      zipAfterPublish: false
+      zipAfterPublish: true
```
Comment (dalemorson) [inline azure-pipelines.yml:67]: You could remove this line as `zipAfterPublish` will be true by default if not present.

File: azure/template.json [hunk 2]
```diff
@@ -11,9 +11,13 @@
...
             "type": "string"
         },
+        "keyvaultCertificateName": {
+            "type": "string",
+            "defaultValue": ""
+        },
         "configNames": {
             "type": "string",
-            "defaultValue": "__SolutionBaseName__"
+            "defaultValue": "__SolutionName__"
```
Comment (jack-cloud-platform) [inline azure/template.json:20]:
```suggestion
            "defaultValue": "SFA.DAS.ApprenticeService.Web"
```
If no config needed in das-employer-config though, this app setting and version aren't needed

File: pipeline-templates/job/code-build.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    - template: azure-pipelines-templates/build/step/app-build.yml@das-platform-building-blocks        
+
+    - template: azure-pipelines-templates/build/step/dependency-check.yml@das-platform-building-blocks
+
+    - task: DotNetCoreCLI@2
+      displayName: "Publish Apprentice Portal Web"
+      inputs:
+        command: publish
+        publishWebProjects: false #set to false as this setting (which defaults to true) will parse the entire repo for web projects
+        projects: src/**/*.csproj
```
Comment (jack-cloud-platform) [inline pipeline-templates/job/code-build.yml:28]:
```suggestion
        projects: src/$(SolutionBaseName)/$(SolutionBaseName).csproj
```
With the SolutionBaseName a variable defined in code-build.yml and SolutionName variable removed from azure-pipelines.yml

File: azure-pipelines.yml [hunk 4]
```diff
@@ -23,61 +21,14 @@ resources:
...
 
-steps:
-  - task: gittools.gittools.setup-gitversion-task.gitversion/setup@0
-    displayName: gitversion/setup
-    inputs:
-      versionSpec: 5.x
+stages:
+- stage: Build
+  jobs:
+  - template: pipeline-templates\job\code-build.yml
```
Comment (jack-cloud-platform) [inline azure-pipelines.yml:27]:
```suggestion
  - template: pipeline-templates/job/code-build.yml
```
Forward slashes so can run on linux as well

File: azure/template.json [hunk 5]
```diff
@@ -44,14 +71,137 @@
...
+                    "keyVaultResourceGroup": {
+                        "value": "[parameters('sharedManagementResourceGroup')]"
+                    }
+                }
             }
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2020-06-01",
+            "resourceGroup": "[variables('resourceGroupName')]",
```
Comment (jack-cloud-platform) [inline azure/template.json:105]: For all resources that are deploying to the resource group, can you add a dependsOn to include the resource group deployment. So the release will work first time.

File: pipeline-templates/job/code-build.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
+jobs:
+- job: CodeBuild
+  pool:
+    name: DAS - Continuous Integration
+  variables:
+  - group: BUILD Management Resources
+  - name: SolutionBaseName
+    value: '__SolutionBaseName__'
```
Comment (jack-cloud-platform) [inline pipeline-templates/job/code-build.yml:8]: Can the value be set here to replace the placeholder

**Review by jack-cloud-platform — APPROVED** [review]

**Pull request merged**